lnwallet/channel.go

```diff
@@ -6584,6 +6627,31 @@ func (lc *LightningChannel) getSignedCommitTx() (*wire.MsgTx, error) {
 	return commitTx, nil
 }
 
+// getSignedCommitTx method takes the latest commitment transaction and
+// populates it with witness data.
+func (lc *LightningChannel) getSignedCommitTx() (*wire.MsgTx, error) {
+	// Fetch the current commitment transaction, along with their signature
+	// for the transaction.
+	localCommit := lc.channelState.LocalCommitment
+
+	inputs := SignedCommitTxInputs{
+		CommitTx:  localCommit.CommitTx,
```

Comment:
Should use `.CommitTx.Copy()` to retain old behavior.

Reply:
Is it really needed? `inputs` is passed to function `GetSignedCommitTx` that does `commitTx := inputs.CommitTx.Copy()` and does not modify `inputs.CommitTx`.

Reply:
I think that functions which modify their arguments (passed through a pointer) should be discouraged. However it is ok for methods like "Unmarshal".